install/install.go

```diff
@@ -660,11 +821,205 @@ func K8sInstaller(c *k8s.Client, o Options) error {
+	}
+
+	f, err := os.Create(filepath.Clean(path.Join(currDir, "kubearmor.yaml")))
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := f.Close(); err != nil {
+			fmt.Printf("Error closing file: %s\n", err)
+		}
+	}()
+
+	for _, o := range printYAML {
+		if err := writeToYAML(f, o); err != nil {
+			return err
+		}
+	}
+
+	file, _ := os.OpenFile("kubearmor.yaml", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+	// Write the string to the file
+	_, err = file.WriteString(manifests + "\n")
+	if err != nil {
+		return err
+	}
+
+	err = f.Sync()
+	if err != nil {
+		log.Fatal(err)
+	}
+	s3 := f.Name()
+	fmt.Println("🤩\tKubeArmor manifest file saved to \033[1m" + s3 + "\033[0m")
 	return nil
 }
 
+func getOperatorConfig(o Options) map[string]interface{} {
+	var operatorImagePullPolicy string = "Always"
+	var opeartorImageTag string = "latest"
+
+	if o.Local {
+		operatorImagePullPolicy = "IfNotPresent"
+	}
+	if o.Tag != "" {
+		opeartorImageTag = o.Tag
+	}
+	if o.ImageRegistry != "" {
+		o.OperatorImage = o.ImageRegistry + "/" + o.OperatorImage
+	}
+	return map[string]interface{}{
+		"kubearmorOperator": map[string]interface{}{
+			"image": map[string]interface{}{
+				"repository": o.OperatorImage,
+				"tag":        opeartorImageTag,
+			},
+			"imagePullPolicy": operatorImagePullPolicy,
+		},
+	}
+}
+
+// K8sInstaller using operator for karmor
+func K8sInstaller(c *k8s.Client, o Options) error {
+	var printYAML []interface{}
+	ns := o.Namespace
+	releaseName := "kubearmor-operator"
+	kubearmorConfig := getOperatorCR(o)
+	values := getOperatorConfig(o)
+	settings := cli.New()
+	settings.SetNamespace(ns)
+
+	actionConfig := actionConfigInit(ns, settings)
+
+	entry := &repo.Entry{
+		Name: "kubearmor",
+		URL:  "https://kubearmor.github.io/charts",
+	}
+
+	r, err := repo.NewChartRepository(entry, getter.All(settings))
+	if err != nil {
+		return fmt.Errorf("failed to create ChartRepository: %w", err)
+	}
+
+	r.CachePath = settings.RepositoryCache
+
+	if _, err := r.DownloadIndexFile(); err != nil {
+		return fmt.Errorf("failed to download repository index file: %w", err)
+	}
+
+	var repoFile repo.File
+	repoFile.Update(entry)
+	if err := repoFile.WriteFile(settings.RepositoryConfig, 0644); err != nil {
+		return fmt.Errorf("failed to write repository file: %w", err)
+	}
+
+	client := action.NewUpgrade(actionConfig)
+	client.Namespace = ns
+	client.Timeout = 5 * time.Minute
+	if o.Save {
+		client.DryRun = true
+		client.SkipCRDs = true
+	}
+
+	chartPath, err := client.ChartPathOptions.LocateChart("kubearmor/kubearmor-operator", settings)
+	if err != nil {
+		return fmt.Errorf("failed to locate Helm chart: %w", err)
+	}
+
+	chartRequested, err := loader.Load(chartPath)
+	if err != nil {
+		return fmt.Errorf("failed to load Helm chart: %w", err)
+	}
+	log.SetOutput(io.Discard)
+	upgradeInstaller, err := client.Run(releaseName, chartRequested, values)
+	log.SetOutput(os.Stdout)
+	if err != nil {
+		client.Install = true
+		if client.Install {
```

Comment:
Is the check needed?

Reply:
`check.Install` is purely an informative flag. However, we pass it to the checkPods function to check whether the snitch has to be inspected or not. The snitch can only be checked at install and not at Helm upgrade, as the snitch job is meant to be erased after installation, leaving no way to track it during upgrades.